### ISSUE TITLE
Need to add oAuth

### ISSUE THREAD

**Issue by iamstuartwilson** [bikedujour/strava-api#1]
@lpww just fleshed out a simple API client at lunch.

Usage:

```javascript
// In script in repo
var stravaApi = require('./')({
    token: 'xxxxxxx'
});

// Get club activities
stravaApi.get('clubs/133132/activities', function(err, data) {
    console.log(data);
});

```

**Comment by iamstuartwilson** [comment]
On second thoughts, you might have access issues to that club.  Try a simple method:

```javascript
stravaApi.get('athlete', function(err, data) {
    console.log(data);
});
```

**Comment by lpww** [comment]
This is awesome, I'll test it out a little later! I'l also look into the oauth handler

**Comment by iamstuartwilson** [comment]
Yeah should be a simple token exchange.  The ingredients will be there.  Some sort of generate login link and then token exchange method should cover it.  The actual oAuth stuff will be handled by the API

**Comment by lpww** [comment]
So it would be something like 

stravaApi.authenticateUser(callback); //returns a login link that calls the callback once authorized

**Comment by iamstuartwilson** [comment]
Probably a bit convoluted.  We'd then have to make a system for handling the login flow too.  We don't want to dictate that, so given the client is initialised with:

```javascript
var stravaApi({
    id: 000,
    secret: 'xxx'
});
```

You need a method that returns an authentication URL with options to set the scope: http://strava.github.io/api/v3/oauth/#get-authorize

Then with the returned code, you need to swap it for an access_token: http://strava.github.io/api/v3/oauth/#post-token

So we don't want to dictate how a user actually gets the code for exchange, just allow them to make that exchange and obtain a permanent access_token!

**Comment by lpww** [comment]
Oh yeah, that solution would only return the temporary token to the user. I forgot that you have to do the exchange. It's the permanent token we want to return!

**Comment by iamstuartwilson** [comment]
Fixed in #3
